core/src/iam/signin.rs

```diff
@@ -370,14 +370,14 @@ mod tests {
 	use std::collections::HashMap;
 
 	#[tokio::test]
-	async fn test_signin_scope() {
+	async fn test_signin_record() {
 		// Test with correct credentials
 		{
 			let ds = Datastore::new("memory").await.unwrap();
 			let sess = Session::owner().with_ns("test").with_db("test");
 			ds.execute(
 				r#"
-				DEFINE SCOPE user SESSION 1h
+				DEFINE ACCESS user ON DATABASE TYPE RECORD DURATION 1h
```

Comment:
There will indeed be breaking changes in `signin`. However, the change in SurrealQL syntax will not be one of them, as the legacy statements (e.g. `DEFINE SCOPE`) will be translated into the new statement (i.e. `DEFINE ACCESS`) during parsing. This ensures that database exports from 1.X can be imported into 2.0.

Of course, this will still require some review to ensure that the business logic remains consistent, as some parameters like `$scope` will no longer exist. Come to think of it, we could keep the `$scope` parameter and return the same as `$access` to preserve this backward compatibility.

In any case, this change is indeed intended to be a breaking one.

Reply:
Nice! That's really cool tbh